src/libs/ajax.js

```diff
@@ -36,117 +36,88 @@ window.saturnMock = {
   } else if (mockResponse) {
-    console.info('%cSimulating response:', consoleStyle)
-    console.info(mockResponse())
-    return new Promise(function(resolve, _) {
-      resolve(mockResponse())
-    })
+    console.info('%cSimulating response:', consoleStyle, mockResponse())
+    return Promise.resolve(mockResponse())
   }
-
-  let withAuth = options
-
-  withAuth.headers = _.merge({
-    'Content-Type': 'application/json',
-    'Authorization': 'Bearer ' + Utils.getAuthToken()
-  }, options.headers)
-
-  return fetch(url, withAuth)
+  return fetch(...args)
 }
 
-const ajaxService = {
-  call(path, success, failure, options) {
-    ajax(`${this.getUrlRoot()}/${path}`, options)
-      .then(response => response.ok ? success(response) : response.text().then(failure))
-      .catch(failure)
-  },
+const fetchOk = (...args) => {
+  return instrumentedFetch(...args)
+    .then(res => res.ok ? res : res.text().then(text => Promise.reject(text)))
+}
 
-  json(path, success, failure, options) {
-    this.call(path, resp => resp.json().then(success), failure, options)
+const Sam = {
+  token: (namespace) => {
+    const scopes = ['https://www.googleapis.com/auth/devstorage.full_control']
+    return fetchOk(`${Config.getSamUrlRoot()}/api/google/user/petServiceAccount/${namespace}/token`, _.merge(authOpts(), bodyOpts(scopes), { method: 'POST' }))
+      .then(parseJson)
   }
 }
 
-const Sam = _.assign({
-  getUrlRoot: Config.getSamUrlRoot,
-
-  token(namespace, success, failure) {
-    const scopes = [
-      'https://www.googleapis.com/auth/devstorage.full_control'
-    ]
-    this.json(`api/google/user/petServiceAccount/${namespace}/token`, success, failure,
-      { method: 'POST', body: JSON.stringify(scopes) })
+export const Buckets = {
+  listNotebooks: (namespace, name) => {
+    return Sam.token(namespace)
+      .then(token => fetchOk(`https://www.googleapis.com/storage/v1/b/${name}/o?prefix=notebooks/`, authOpts(token)))
+      .then(parseJson)
+      .then(({ items }) => _.filter(items, ({ name }) => name.endsWith('.ipynb')))
   }
-}, ajaxService)
-
-export const Buckets = _.assign({
-  getUrlRoot: () => 'https://www.googleapis.com/storage/v1/b',
-
-  listNotebooks(namespace, name, success, failure) {
-    Sam.token(namespace,
-      token => {
-        this.json(`${name}/o?prefix=notebooks/`,
-          res => success(_.filter(res.items, item => item.name.endsWith('.ipynb'))),
-          failure,
-          { headers: { Authorization: 'Bearer ' + token } })
-      },
-      failure)
-  }
-}, ajaxService)
-
+}
 
-export const Rawls = _.assign({
-  getUrlRoot: () => `${Config.getRawlsUrlRoot()}/api`,
+const fetchRawls = (path, ...args) => fetchOk(`${Config.getRawlsUrlRoot()}/api${path}`, ...args)
 
-  workspacesList(success, failure) {
-    this.json('workspaces', success, failure)
+export const Rawls = {
+  workspacesList: () => {
+    return fetchRawls('/workspaces', authOpts())
```

Comment:
Any strong reason to have the slash here rather than in the url template string?